pl-faded-parsons.py

```diff
@@ -1,314 +1,538 @@
+    `file-name`  
+        The file to store the student's submission for grading. Defaults to `user_code.py`.  
+    `solution-path`  
+        The path to a file containing the solution. Defaults to "./solution".  
+
+    Attributes
+    ----------
+    `answers_name` : `str`
+        This problem's identifier. Specified with `answers-name="..."`.
+    `format` : `FadedParsonsProblem.Formats`
+        The provided format of the problem. Specified with `format="..."`.
+    `markup` : `str`
+        The markup provided in html that is parsed into lines for the student. Is not used if the student has previously made a submission.
+    `pre_text` : `str`
+        The text that will be shown directly before the solution tray. Will be an empty string and not rendered if omitted. Cannot be used with format="right".
+    `post_text` : `str`
+        The text that will be shown directly after the solution tray. Will be an empty string and not rendered if omitted. Cannot be used with format="right".
+    `language` : `str`
+        The language with which to apply syntax highlighting. May be an empty string, in which case no highlighting will be done.
+    `out_filename` : `str`
+        The file to which to include the student's submission. Specified with `file-name="..."`.
+    `size` : `Literal["narrow", "wide"]`
+        The size of the solution tray. `"narrow"` indicates it should take approximately half the width of the problem pane. `"wide"` indicates it should take the full width of the problem pane.
+    `solution_path` : `str`
+        The path to a file containing the solution. Specified with `solution-path="..."`. Raises `FileNotFoundError` on access if not found.
+    `solution` : `str`
+        The solution. Specified with `solution-path="..."`. Raises `FileNotFoundError` on access if file not found.
+    `trays` : `Submission.Trays`
+        The trays used in this problem. MUST CALL `.load(...)` TO DEFINE.
+    `log` : `List[Submission.LogEntry]`
+        The log of events for this problem. MUST CALL `.load(...)` TO DEFINE.
+
+    Methods
+    -------
+    `to_mustache() -> Mustache`
+        Produce a `Mustache` instance for rendering
+    `to_code() -> str`
+        Compile the student submission into an executable code snippet.
+
+    """
+
+    class Formats(Enum):
+        BOTTOM = "bottom"
+        RIGHT = "right"
+        NO_CODE = "no_code"
+
+    @property
+    def solution_path(self) -> str:
+        if not os.path.exists(self._solution_path):
+            raise FileNotFoundError('\n'
+                f'\tCorrect answer not found at `{self._solution_path}`! \n'
+                 '\tProvide an answer or set "showCorrectAnswer" to false in `./info.json`'
+            )
+
+        return self._solution_path
+
+    @property
+    def solution(self) -> str:
+        with open(self.solution_path, "r") as f:
+            return f.read()
+
+    def __init__(self, element_html: str, data: pl.QuestionData):
```

Comment:
If we could prevent `pl` from being required at the top level (or stub it), that would be great.

The idea behind the try/import at the top is so that the file can be loaded locally in venv without a PL instance running; as is, the type def causes python/ipython to crash on load.